test/activation_based/test_distributed_planning.py

```
@@ -0,0 +1,139 @@
+
+def test_planner_does_not_assume_fp8_hybrid_memory_savings():
+    config = _config()
+    config.sequence_length = 127
+    config.model.transformer.hidden_size = 120
+    config.model.transformer.ffn_hidden_size = 480
+    config.model.transformer.num_attention_heads = 3
+    config.model.transformer.num_query_groups = 3
+    config.model.transformer.kv_channels = 40
+    config.model.transformer.fp8 = "hybrid"
+
+    with pytest.raises(ValueError, match="does not fit|No topology fits"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)